usage of taxa thesaurus family over occurrence family

MuchQuak opened the pull request:
# Pull Request Checklist:

# Pre-Approval

- [x] There is a description section in the pull request that details what the proposed changes do. It can be very brief if need be, but it ought to exist.
- [x] Hotfixes should be branched off of the `master` branch and **squash and merged** back into the `master` branch.
- [x] Features and backlog bugs should be merged into the `Development` branch, **NOT** `master`
- [x] All new text is preferably internationalized (i.e., no end-user-visible text is hard-coded on the PHP pages), and the [spreadsheet tracking internationalizations](https://docs.google.com/spreadsheets/d/133fps9w2pUCEjUA6IGCcQotk7dn9KvepMXJ2IWUZsE8/edit?usp=sharing) has been updated either with a new row or with checkmarks to existing rows.
- [x] There are no linter errors
- [x] New features have responsive design (i.e., look aesthetically pleasing both full screen and with small or mobile screens)
- [x] [Symbiota coding standards](https://docs.google.com/document/d/1-FwCZP5Zu4f-bPwsKeVVsZErytALOJyA2szjbfSUjmc/edit?usp=sharing) have been followed
- [x] If any files have been reformatted (e.g., by an autoformatter), the reformat is its own, separate commit in the PR
- [x] Comment which GitHub issue(s), if any does this PR address
- [x] If this PR makes any changes that would require additional configuration of any Symbiota portals outside of the files tracked in this repository, make sure that those changes are detailed in [this document](https://docs.google.com/document/d/1T7xbXEf2bjjm-PMrlXpUBa69aTMAIROPXVqJqa2ow_I/edit?usp=sharing).

# Post-Approval

- [x] It is the code author's responsibility to merge their own pull request after it has been approved
- [x] If this PR represents a merge into the `Development` branch, remember to use the **squash & merge** option
- [x] If this PR represents a merge from the `Development` branch into the master branch, remember to use the **merge** option
- [x] If this PR represents a hotfix into the `master` branch, a subsequent PR from `master` into `Development` should be made **merge** option (i.e., no squash).
- [x] If the dev team has agreed that this PR represents the last PR going into the `Development` branch before a tagged release (i.e., before an imminent merge into the master branch), make sure to notify the team and [lock the `Development` branch](https://github.com/BioKIC/Symbiota/settings/branches) to prevent accidental merges while QA takes place. Follow the release protocol [here](https://github.com/BioKIC/Symbiota/blob/master/docs/release-protocol.md).
- [x] Don't forget to delete your feature branch upon merge. Ignore this step as required.

Thanks for contributing and keeping it clean!

MuchQuak committed:
usage of taxa thesaurus family over occurrence family

themerekat commented:
Seems to be working, but the scientific names are not in order
![image](https://github.com/user-attachments/assets/78db2aee-cd60-4d37-be11-0d42f9d0a54d)

themerekat reviewed (approved):
Works great!

egbot committed:
Modifications

- "ts.taxauthid = 1" needs to be added to the where condition or the recordset would return duplicate records for each taxonomic thesaurus defined within taxstatus table (e.g. there are 4 defined within SEINet). I suppose not including it would not be a big problem since the occid would be the same for each record, and thus the output data array would just copy over existing data. However, it would iterate through more records than needed. Another solution would be to change the statement to "SELECT DISTINCT...", but this can significantly slows down an ordered query.
- Since "ts.taxauthid = 1" is added to the where clause, the getTableJoins function should always add the taxstatus left join, thus we can leave out that test.

egbot committed:
Minor fix to previous submit

MuchQuak commented:
> > Hold-on, I'm testing something. We might need to adjust this. More details soon.
> 
> I was worried about situations where portals have multiple taxonomic thesaurus installed. The general rule is that anytime you join to the taxstatus or taxaenumtree tables, you need to add taxauthid = 1 (or such) to the condition. Otherwise, you will increase your records output by a factor roughly equivalent to the number of thesauri defined. See commit #1675 for my suggested solution. I'm now thinking that it might not have been as big of an issue as I originally thought, but it's still best to limit taxauthid. See commit description for details.

Gotcha wasn't really sure what the edge cases around the thesaurus were. I will keep that in mind when working on other features related to that.
